cinema/views.py

```diff
@@ -1,45 +1,115 @@
-from rest_framework.decorators import api_view
+from django.http import HttpRequest, HttpResponse, Http404
 from rest_framework.response import Response
-from rest_framework import status
+from rest_framework import status, generics, mixins, viewsets
 
 from django.shortcuts import get_object_or_404
+from rest_framework.views import APIView
 
-from cinema.models import Movie
-from cinema.serializers import MovieSerializer
+from cinema.models import Actor, Genre, CinemaHall, Movie
+from cinema.serializers import (MovieSerializer,
+                                ActorSerializer,
+                                GenreSerializer,
+                                CinemaHallSerializer,)
 
 
-@api_view(["GET", "POST"])
-def movie_list(request):
-    if request.method == "GET":
-        movies = Movie.objects.all()
-        serializer = MovieSerializer(movies, many=True)
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+    serializer_class = MovieSerializer
+
+
+class GenreList(APIView):
+    def get(self, request: HttpRequest) -> HttpResponse:
+        genres = Genre.objects.all()
+        serializer = GenreSerializer(genres, many=True)
+
         return Response(serializer.data, status=status.HTTP_200_OK)
 
-    if request.method == "POST":
-        serializer = MovieSerializer(data=request.data)
+    def post(self, request: HttpRequest) -> HttpResponse:
+        serializer = GenreSerializer(data=request.data)
+
         if serializer.is_valid():
             serializer.save()
             return Response(serializer.data, status=status.HTTP_201_CREATED)
 
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
-@api_view(["GET", "PUT", "DELETE"])
-def movie_detail(request, pk):
-    movie = get_object_or_404(Movie, pk=pk)
+class GenreDetail(APIView):
+    def get_object(self, pk: int) -> Genre:
+        return get_object_or_404(Genre, pk=pk)
 
-    if request.method == "GET":
-        serializer = MovieSerializer(movie)
-        return Response(serializer.data, status=status.HTTP_200_OK)
+    def get(self, request: HttpRequest, pk: int) -> HttpResponse:
+        genre = self.get_object(pk=pk)
+        serializer = GenreSerializer(genre)
+
+        return Response(serializer.data)
```

Comment:
optional, all status codes are set explicitly in your code, so I'd add them in other places here as well